code/modules/mob/living/simple_animal/hostile/winter_mobs.dm

```diff
@@ -0,0 +1,141 @@
+	name = "reindeer"
+	desc = "Apparently murder is a reindeer game."
+	icon_state = "reindeer"
+	icon_living = "reindeer"
+	icon_dead = "reindeer-dead"
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat
+	meat_amount = 3
+
+	melee_damage_lower = 5
+	melee_damage_upper = 10
+
+/mob/living/simple_animal/hostile/winter/santa
+	maxHealth = 150		//if this seems low for a "boss", it's because you have to fight him multiple times, with him fully healing between stages
+	health = 150
+	var/next_stage = null
+	var/death_message
+	name = "Santa Claus"
+
+	icon_state = "santa"
+	icon_living = "santa"
+	icon_dead = "santa-dead"
+
+/mob/living/simple_animal/hostile/winter/santa/death()
+	..()
+	if(death_message)
+		visible_message(death_message)
+	if(next_stage)
+		spawn(10)
+			new next_stage(get_turf(src))
+			qdel(src)	//hide the body
+
+/mob/living/simple_animal/hostile/winter/santa/stage_1		//stage 1: slow melee
+	desc = "GET THE FAT MAN!"
+	next_stage = /mob/living/simple_animal/hostile/winter/santa/stage_2
+	death_message = "<span class='danger'>HO HO HO! YOU THOUGHT IT WOULD BE THIS EASY?!?</span>"
+	speed = 2
+	melee_damage_lower = 5
+	melee_damage_upper = 15
+
+/mob/living/simple_animal/hostile/winter/santa/stage_2		//stage 2: slow ranged
+	desc = "GET THE FAT MAN AGAIN!"
+	next_stage = /mob/living/simple_animal/hostile/winter/santa/stage_3
+	death_message = "<span class='danger'>YOU'VE BEEN VERY NAUGHTY! PREPARE TO DIE!</span>"
+	ranged = 1
+	projectiletype = /obj/item/projectile/ornament
+	retreat_distance = 5
+	minimum_distance = 5
+
+/mob/living/simple_animal/hostile/winter/santa/stage_3		//stage 3: fast rapidfire ranged
+	desc = "WHY WON'T HE DIE ALREADY!?"
+	next_stage = /mob/living/simple_animal/hostile/winter/santa/stage_4
+	death_message = "<span class='danger'>FACE MY FINAL FORM AND KNOW DESPAIR!</span>"
+	ranged = 1
+	rapid = 1
+	speed = 0	//he's lost some weight from the fighting
+	projectiletype = /obj/item/projectile/ornament
+	retreat_distance = 3
+	minimum_distance = 3
+
+/mob/living/simple_animal/hostile/winter/santa/stage_4		//stage 4: fast spinebreaker
+	name = "Final Form Santa"
+	desc = "WHAT THE HELL IS HE!?! WHY WON'T HE STAY DEAD!?!"
+	ranged = 0
+	rapid = 0
+	speed = 0	//he's lost some weight from the fighting
+
+	environment_smash = 2		//naughty walls must be punished too
+	melee_damage_lower = 10
+	melee_damage_upper = 25		//that's gonna leave a mark, for sure
+
+/mob/living/simple_animal/hostile/winter/santa/stage_4/death()
+	world << "<span class='notice'><hr></span>"
+	world << "<span class='notice'>THE FAT MAN HAS FALLEN!</span>"
+	world << "<span class='notice'>SANTA CLAUS HAS BEEN DEFEATED!</span>"
+	world << "<span class='notice'><hr></span>"
+	..()
+	var/obj/item/weapon/grenade/clusterbuster/xmas/X = new /obj/item/weapon/grenade/clusterbuster/xmas(get_turf(src))
+	var/obj/item/weapon/grenade/clusterbuster/xmas/Y = new /obj/item/weapon/grenade/clusterbuster/xmas(get_turf(src))
+	X.prime()
+	Y.prime()
```

Comment:
This makes a LOT of presents by the way. Fat man was definitely packing some serious loot.